File: go/src/sorts/quicksort_test.go
package sorts

import "../testhelper"

import "testing"

func TestQuicksort(t *testing.T) {
  arr := testhelper.RandomIntSlice(testhelper.ArraySize)

  sorted := Quicksort(arr)

  testhelper.AssertIsSorted(sorted, t)
}

File: python/structures/test/sorts/quick_sort_test.py
import structures.src.sorts.quick as qsort
from structures.src.util.sort_util import is_sorted, random_array
import structures.src.util.constants as const


def test_sorted():
    arr = random_array(const.ARRAY_SIZE)

    # make sure the starting array isn't sorted
    assert not is_sorted(arr)

    # sort the array
    result = qsort.quicksort(arr)

    # make sure it worked
    assert is_sorted(result)
